app/controllers/vendors_controller.rb

```diff
@@ -36,8 +38,25 @@ def create
 
   def update
     @vendor = Vendor.find(params[:id])
-    Vendor.approve(@vendor)
-    flash[:notice] = "Vendor Updated"
+    if params[:_method] == "put"
+      Vendor.approve(@vendor)
+      flash[:notice] = "Vendor Updated"
+    elsif params[:_method] == "patch"
+      if params[:commit] == "Claim Business"
+        if @vendor.claimed_status == "Unclaimed"
+          @vendor.update(owner_id: current_user.id, claimed_status: "Pending")
+          flash[:notice] = "Submitted for approval."
+        elsif @vendor.claimed_status == "Pending"
+          flash[:notice] = "This business is pending a previous ownership claim"
+        end
+      elsif params[:commit] == "Approve Ownership"
+        @vendor.update(claimed_status: "Claimed")
+        flash[:notice] = "Owner approved."
+      elsif params[:commit] == "Deny Claim"
+        @vendor.update(claimed_status: "Unclaimed")
+        flash[:notice] = "Owner denied."
+      end
+    end
```

Comment:
Ohhh man this is a gigantic action! I think this has gotten a bit too bloated to allow it to live on in its current state. Is there anything here you could extract to helper methods? Also, renaming things to be a little more descriptive would be good - I don't know as a reader what the differences are between how/when/why you're having `:_method` be `put` vs `patch, even just finding a more descriptive way to differentiate between those two big if/else clauses would be good